tuf/fetcher.py

```diff
@@ -113,12 +113,11 @@ def fetch(self, url, required_length):
     # Always set the timeout. This timeout value is interpreted by requests as:
     #  - connect timeout (max delay before first byte is received)
     #  - read (gap) timeout (max delay between bytes received)
-    with session.get(url, stream=True,
-        timeout=self.SOCKET_TIMEOUT) as response:
-
-      # Check response status.
-      response.raise_for_status()
+    response = session.get(url, stream=True, timeout=self.SOCKET_TIMEOUT)
```

Comment:
As there's no longer a context manager in use for `response`, do we need to call some cleanup method?

Reply:
also maybe double check what happens when connect times out. I expect that's a ReadTimeoutError which should be handled

Reply:
Did some cleanup in 77900af.

Connect time out will raise an exception that will be captured in `NoWorkingMirrorError`. This is the same as the current behaviour.